src/Java.Interop/Java.Interop/JniRuntime.cs

```diff
@@ -88,6 +96,7 @@ public partial class JniRuntime : IDisposable
 		const   int     JNI_EDETACHED   = -2;
 		const   int     JNI_EVERSION    = -3;
 
+		static public bool JvmDllLoaded  { get; private set; }
```

Comment:
Should this actually be `static`?

I also don't see why this needs to be part of the *public* API surface.  `protected` I could understand, as it's used by `JreRuntime`, but no `public` access should need to know this.

Reply:
...ditto `JvmLibraryLoaded`.

Reply:
That said, I don't like the semantics here at all.

The semantics appear to be "if the JVM library was loaded, use `java-interop!java_interop_jvm_list()`, otherwise use `jvm!JNI_GetCreatedJavaVMs()`.

The problem here is that we *know* 99% of the time `jvm!JNI_GetCreatedJavaVMs()` has *no* chance of working, because `JVM.DLL` won't be findable; it won't be in `%PATH%` (Windows), unless we take care to add it there (blech), and it won't be in `Java.Runtime.Environment.dll.config` on *end user machines*, because the actual path to the JVM is unknowable before installation (and will likely be invalidated once a Java security update is applied). (Plus, `Java.Runtime.Environment.dll.config` won't be world writable, so "normal user processes" won't be able to update it, even if they did have a newer location.)

What I'm driving at is that supporting `jvm!JNI_GetCreatedJavaVMs()` doesn't make sense to me.

I'd rather drop it entirely, and *require* use of `java-interop!java_interop_jvm_list()`. That is a library we fully control, and can ensure is in a findable location.

Reply:
I kept the original code path (calls to `JNI_GetCreatedJavaVMs` and `JNI_CreatedJavaVM`) for the tests, so that we don't need to update them to use the new code.

Reply:
> Should this actually be static?

I think you mean `public` here

> I also don't see why this needs to be part of the public API surface. protected I could understand, as it's used by JreRuntime, but no public access should need to know this.

Indeed. I though I made it protected, but looks like I did not.

Reply:
> I kept the original code path (calls to JNI_GetCreatedJavaVMs and JNI_CreatedJavaVM) for the tests, so that we don't need to update them to use the new code.

And also for Android. It is used in `Java.Interop/JniRuntime.cs`, so it looks like it is not desktop specific.